### DISCUSSION TITLE
Controlling image size

### DISCUSSION THREAD

**Discussion by dransome** [cerb/cerb-release#60] [Q&A] [answered]
When pasting images (e.g. screenshots) into ticket replies, they are sometimes very large. Is there a way to control this within Cerb itself?

E.g. in gmail and some other systems there's an option to select between original, best fit, small, medium, large etc. for any pasted image. Presumably this coerces the image via width constraints in the underlying html.

Maybe there's a way to set a max-width value somewhere in Cerb config / templating?

**Comment by jstanden** [comment, marked as answer]
Hi @dransome!

On recent versions of Cerb, inline images for HTML email should have a `max-width: 100%` so they scale to the browser window.

If you're having issues with what a recipient sees in their email client, that would be controlled by the HTML email template (Search -> Email Templates).

The mail template can define a stylesheet for how the message should be rendered.  Make sure you have something like:

```css
img {
  max-width: 100%;
}
```

That's a default for any new mail templates, but it may not be included for older pre-existing templates. You can just add it.

If you're looking for a way to resize the actual underlying file, we don't currently provide that by default but it could be implemented with an interaction in the reply toolbar.

You can get the dimensions of an image with:
https://cerb.ai/docs/bots/scripting/filters/#image_info

We don't currently have image manipulation functions, but there's a feature branch in 10.5 for creating + modifying images.

In the meantime you could use `http.request:` <https://cerb.ai/docs/automations/commands/http.request/> and an API like: https://tinypng.com/developers

You'd send the image bytes like this:
https://cerb.ai/docs/automations/commands/http.request/#stream-a-large-upload-from-an-attachment

On macOS I highly recommend Cleanshot: https://cleanshot.com

You can quickly modify your screenshots with Cleanshot before pasting them -- resizing, cropping, adding callouts, arrows, etc.
